src/main/java/frc/team2412/robot/commands/intake/IntakeBitmapCommand.java

```diff
@@ -0,0 +1,98 @@
+        B(true, false, true, false, INTAKE_IN_SPEED, INDEX_IN_SPEED, INDEX_IN_SPEED, false), // correct ball in ingest
+        C(true, false, false, false, INTAKE_IN_SPEED, INDEX_IN_SPEED, INDEX_IN_SPEED, false), // wrong ball in ingest
+        D(false, true, false, true, INTAKE_IN_SPEED, INDEX_IN_SPEED, 0, false), // correct ball in feeder
+        E(false, true, false, false, INTAKE_IN_SPEED, INDEX_IN_SPEED, INDEX_IN_SPEED, true), // wrong ball in feeder
+        F(true, true, true, true, 0, 0, 0, false), // correct ball in both
+        G(true, true, false, true, INTAKE_OUT_SPEED, INDEX_OUT_SPEED, 0, false), // wrong ball in ingest, corect ball in
+                                                                                    // shooter (may change later)
+        H(true, true, true, false, 0, INDEX_IN_SPEED, INDEX_IN_SPEED, true), // correct ball in ingest, wrong ball in
+                                                                                // feeder
+        I(true, true, false, false, 0, INDEX_IN_SPEED, INDEX_IN_SPEED, true); // wrong ball in both
+
+        boolean ingestSensor, feederSensor, ingestColor, feederColor, shooterMisfire;
+        double intakeMotorSpeed, ingestMotorSpeed, feederMotorSpeed;
+
+        private Bitmap(boolean ingestSensor, boolean feederSensor, boolean ingestColor, boolean feederColor,
+                double intakeMotorSpeed, double ingestMotorSpeed, double feederMotorSpeed, boolean shooterMisfire) {
+            this.ingestSensor = ingestSensor;
+            this.feederSensor = feederSensor;
+            this.ingestColor = ingestColor;
+            this.feederColor = feederColor;
+            this.intakeMotorSpeed = intakeMotorSpeed;
+            this.ingestMotorSpeed = ingestMotorSpeed;
+            this.feederMotorSpeed = feederMotorSpeed;
+            this.shooterMisfire = shooterMisfire;
+        }
+
+        public boolean equals(boolean ingestSensor, boolean feederSensor, boolean ingestColor, boolean feederColor) {
+            return this.ingestSensor == ingestSensor &&
+                    this.feederSensor == feederSensor &&
+                    this.ingestColor == ingestColor &&
+                    this.feederColor == feederColor;
+        }
+
+    }
+
+    private IntakeSubsystem intakeSubsystem;
+    private IndexSubsystem indexSubsystem;
+    private ShooterSubsystem shooterSubsystem;
+    private ShooterVisionSubsystem shooterVisionSubsystem;
+
+    public IntakeBitmapCommand(IntakeSubsystem intakeSubsystem, IndexSubsystem indexSubsystem,
+            ShooterSubsystem shooterSubsystem, ShooterVisionSubsystem shooterVisionSubsystem) {
+        this.intakeSubsystem = intakeSubsystem;
+        this.indexSubsystem = indexSubsystem;
+        this.shooterSubsystem = shooterSubsystem;
+        this.shooterVisionSubsystem = shooterVisionSubsystem;
+    }
+
+    @Override
+    public void execute() {
```

Comment:
This is a method with logic we will want to optimize and understand.  Please add some logging to help us troubleshoot and understand state.